gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/FsBenchmark.java

```diff
@@ -156,24 +195,38 @@ private int benchmarkRead(FileSystem fs, Map<String, String> args) {
     warmup(
         args,
         () ->
-            benchmarkRead(
-                fs, testFile, /* readSize= */ 1024, /* numReads= */ 1, /* numThreads= */ 2));
+            benchmarkFileIO(
+                fs,
+                testFile,
+                OperationType.READ,
+                /* readSize= */ 1024,
+                /* numReads= */ 1,
+                /* numThreads= */ 2,
+                /* totalFileSize */ 0));
 
-    benchmarkRead(
+    benchmarkFileIO(
         fs,
         testFile,
+        OperationType.READ,
         parseInt(args.getOrDefault("--read-size", String.valueOf(1024))),
         parseInt(args.getOrDefault("--num-reads", String.valueOf(1))),
-        parseInt(args.getOrDefault("--num-threads", String.valueOf(1))));
+        parseInt(args.getOrDefault("--num-threads", String.valueOf(1))),
+        0);
 
     return 0;
   }
 
-  private void benchmarkRead(
-      FileSystem fs, Path testFile, int readSize, int numReads, int numThreads) {
+  private void benchmarkFileIO(
```

Comment:
looks like we are adding to the complexity of the method, please consider refactoring